### PR TITLE
TECH-1169 Generic deployment to environment Composite Action

### PR DESCRIPTION
**Pull request by tonisojandu-sympower**
* Renamed `deploy-to-staging` to `deploy-to-evnrionment`. 
* Added `deploy-group` field to give more control over what `auto-deploy/**.env.json` files will be used in the deployment. If `deployGroup` field is not filled in the JSON file then uses `environment` field as `deployGroup` value for simplicity and backwards compatibility.

[Succesful experimentation here](https://github.com/sympower/msa-observability/actions/runs/6653343744).

### CODE REVIEW

File: deploy-to-environment/action.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+name: 'Deploy to environment'
+description: 'Upgrade the environment with the latest docker image'
+inputs:
+  secrets:
+    description: 'Secrets required for the build'
+    required: true
+    default: '{}'
```
Comment (florianbuth-sympower) [inline deploy-to-environment/action.yml:7]: No default needed if required 🙂

File: deploy-to-environment/action.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  version:
+    description: 'Version number of the image released to ECR'
+    required: true
+  default-name:
```
Comment (florianbuth-sympower) [inline deploy-to-environment/action.yml:11]: This is unused?

Comment (tonisojandu-sympower) [reply deploy-to-environment/action.yml:10]: Fixed it now.

File: deploy-to-environment/action.yml (hunk 1 above)
Comment (tonisojandu-sympower) [reply deploy-to-environment/action.yml]: True. Removed default from secrets in all composite actions.

**Review by florianbuth-sympower — APPROVED** [review]
Looks good to me

**Pull request merged**